src/arrayMethodJoin.js

```diff
@@ -4,8 +4,28 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    const copyArr = [...this];
+    let arrJoin = '';
+    let sep = separator;
+
+    if (sep === null) {
+      sep = 'null';
+    }
+
+    for (let i = 0; i < copyArr.length; i++) {
+      if (copyArr[i] === null || copyArr[i] === undefined) {
+        copyArr[i] = '';
+      }
```

Comment:
```suggestion
      if (this[i] !== null && this[i] !== undefined) {
        arrJoin += this[i];
      }
```
Just do nothing if element is null or undefined